### PR TITLE
carver: tar creation is now streamed

### PR DESCRIPTION
**Pull request by muffins**
This brings a "streaming" file IO to the carver for the tar archive creation to ensure that large carves do not bump up against our file read limits, as well as adds a few small checks to enhance the user experience of the carver.

### CODE REVIEW

**Comment by muffins** [comment]
ok to test

**Comment by muffins** [comment]
ok to test

**Comment by muffins** [comment]
ok to test

File: osquery/carver/carver.cpp [hunk 1]
```diff
@@ -229,24 +232,34 @@ Status Carver::compress(const std::set<boost::filesystem::path>& paths) {
...
   archive_write_free(arch);
 
   PlatformFile archFile(archivePath_.string(), PF_OPEN_EXISTING | PF_READ);
   updateCarveValue(carveGuid_, "size", std::to_string(archFile.size()));
-  updateCarveValue(
-      carveGuid_,
-      "sha256",
-      hashFromFile(HashType::HASH_TYPE_SHA256, archivePath_.string()));
 
+  std::string arcHash =
```
Comment (jedi22) [inline osquery/carver/carver.cpp]: This will need documentation

Comment (muffins) [inline osquery/carver/carver.cpp]: Good call!

File: osquery/carver/carver.cpp [hunk 2]
```diff
@@ -147,15 +149,16 @@ void Carver::start() {
...
-      }
+    // Ensure the file is a flat file on disk before "carving"
+    PlatformFile pFile(p.string(), PF_OPEN_EXISTING | PF_READ);
+    if (!pFile.isValid() || isDirectory(p)) {
+      VLOG(1) << "File does not exist on disk or is subdirectory: " << p;
+      continue;
+    }
+    Status s = carve(p);
+    if (!s.ok()) {
+      VLOG(1) << "Failed to carve file: " << p;
```
Comment (jedi22) [inline osquery/carver/carver.cpp]: Can we also get the reason from the status here?

**Comment by osqueryer** [comment]
:-1: The commit 3d2821c4bd66608c742a8d295f10fb8c98e90309 (Job results: [1322](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1322/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit 3d2821c4bd66608c742a8d295f10fb8c98e90309 (Job results: [1323](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1323/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit 3d2821c4bd66608c742a8d295f10fb8c98e90309 (Job results: [2105](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2105/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 3d2821c4bd66608c742a8d295f10fb8c98e90309 (Job results: [2106](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2106/)) failed one or more tests (**Windows**).

**Comment by muffins** [comment]
ok to test

**Pull request merged**